### PR TITLE
MET-5053 Add Pop-Out Component / Record Links Refactor

### PR DESCRIPTION
**Pull request by andyjmaclean**

### CODE REVIEW

**Review by jeortizquan — COMMENTED** [review]
I can see the component and helps to the user.
image
It simplified the amount of code.

image
and as we hover shows the panel.

There is one observation, is this intended?
image
when I click tier 0 it disappears and this bar stays static there
image

**Comment by andyjmaclean** [comment]
No this is not intended!  I thought I'd fixed that!

File: projects/sandbox/src/app/problem-viewer/problem-viewer.component.html [hunk 1]
```diff
@@ -412,67 +412,51 @@ <h3 class="problem-viewer" *ngIf="ppd.problemPatternList.length === 0; else prob
...
-              >
-              <sb-copyable-link-item
-                [labelRef]="labelRefPortal"
-                [href]="processedRecordData ? processedRecordData.portalRecordLink : ''"
-              ></sb-copyable-link-item>
-            </li>
-          </ul>
-        </div>
-      </div>
+      <sb-pop-out
```
Comment (jeortizquan) [inline projects/sandbox/src/app/problem-viewer/problem-viewer.component.html:415]: does this require a name or id?

File: projects/sandbox/src/app/progress-tracker/progress-tracker.component.html [hunk 2]
```diff
@@ -108,40 +108,21 @@
...
-      class="warning-view"
-      *ngIf="
-        (tierInfo['content-tier'] && tierInfo['content-tier'].total) ||
-        (tierInfo['metadata-tier'] && tierInfo['metadata-tier'].total)
-      "
-      [ngClass]="{
-        closed: !warningViewOpen
-      }"
+    <sb-pop-out
+      #tierPopOut
```
Comment (jeortizquan) [inline projects/sandbox/src/app/progress-tracker/progress-tracker.component.html:114]: is this the name or id of the component?

**Review by jeortizquan — APPROVED** [review]
I checked and the functionality is working now.

File: projects/sandbox/src/app/progress-tracker/progress-tracker.component.html (hunk 2 above)
Comment (andyjmaclean) [reply projects/sandbox/src/app/progress-tracker/progress-tracker.component.html:114]: It's a local reference used to handle clicks on the title.  It's a more lightweight way of accomplishing what could be done with a ViewChild reference.

File: projects/sandbox/src/app/problem-viewer/problem-viewer.component.html (hunk 1 above)
Comment (andyjmaclean) [reply projects/sandbox/src/app/problem-viewer/problem-viewer.component.html:415]: No.  The pop-out for the record-links is simpler and does not contain the title (close) link

**Pull request merged**